.github/workflows/js-build-and-lint.yml

```diff
@@ -55,14 +59,34 @@ jobs:
           echo "::add-matcher::.github/tsc.json"
           echo "::add-matcher::.github/eslint-stylish.json"
 
-      - name: eslint + fix
-        uses: getsentry/action-eslint-fix@v2
-        if: steps.changes.outputs.frontend == 'true'
-        with:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      # Lint entire frontend if this is on main branch
+      - name: eslint
+        if: github.ref == 'refs/heads/master' && steps.changes.outputs.frontend == 'true'
+        run: |
+          # Run relax config on main branch (and stricter config for changed files)
+          yarn lint -c .eslintrc.relax.js
+          yarn lint:css
+
+      # Otherwise if it's not main branch, only lint modified files
+      - name: eslint (changed files only)
+        if: github.ref != 'refs/heads/master' && steps.changes.outputs.frontend == 'true'
+        run: |
+          yarn eslint --fix ${{ steps.changes.outputs.frontend_modified_lintable_files }}
+
+      # Otherwise if it's not main branch, only lint modified files
+      - name: Commit any eslint fixed files
+        if: github.ref != 'refs/heads/master' && steps.changes.outputs.frontend == 'true'
+        continue-on-error: true
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions@sentry.io"
+          git add -A
+          git commit -m "style(): Auto eslint fix ${GITHUB_SHA}" || exit 0
+          git push origin "HEAD:${{ github.event.pull_request.head.ref }}"
```

Comment:
Is there a need to be so explicit here?

Reply:
There was but I changed the checkout to checkout b ranch instead of a commit so we can remove this.